### PR TITLE
Add es6 build

### PR DESCRIPTION
**Pull request by TPRobots**
Pull request opened by github-pullrequestcreator.

- Generate library using typescript and change `main` from `package.json` to point to it. It stills creates a dev and prod bundle accessible in the `dist` folder.
- Remove UI from production bundle so `css`, `images` and ui logic are included. If z`ui.show` or `ui.hide` are called in production, a placeholder message will be displayed instead.
  - Before: prod => 25.3kb
  - After: prod => 18kb
  - (bundle decreased by around 30%)

### CODE REVIEW

**Comment by nealstewart** [comment]
Just use an async import of the UI in both development and production:

https://webpack.js.org/api/module-methods/#import-

**Comment by nealstewart** [comment]
That will still allow the UI, but only enable it if you load it

**Comment by nealstewart** [comment]
I misunderstood the build change. Ignore that, but consider doing the async import change

**Comment by thomasthiebaud** [comment]
It's a library, so you want to be able to import modules and let your own bundler (webpack or whatever) take care of minifiying it. This is quite a standard way to export a library.

Excluding the UI from production is a proposal and we can discuss it. By doing so we make sure that a customer will never see this UI.

File: src/ui.prod.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+export const uiFactory = (
+    tests: BehavioralSubject<SplitTest[]>,
+    reset: () => void,
+    getCurrentTestVariation: (testName: string) => string,
+    getUserAgentInfo: () => UserAgentInfo,
+    setCurrentTestVariation: (testName: string, variation: string) => void,
+) => {
+    function show() {
+        console.log('[Skift] Running in production, UI is disabled');
```
Comment (nealstewart) [inline src/ui.prod.ts]: Could you make this log the variations and current variation in the console?

**Pull request merged**